### DISCUSSION TITLE
Rust reqwest

### DISCUSSION THREAD

**Discussion by stevenj** [Kong/httpsnippet#278] [New Client Request]
Rust would be a nice addition to the library.

**Comment by philsturgeon** [comment]
@stevenj have you had a look at how the targets work? It's pretty clever, it's JavaScript that builds up a string. Maybe you could create the best Rust target in pure Rust, pop it here as step 1, then try and code it up in JavaScript as step 2.

**Example:** https://github.com/Kong/httpsnippet/blob/master/src/targets/go/native.js

**Comment by fahadahammed** [comment]
I am missing RUST too... :(

**Comment by dimitropoulos** [comment]
If anyone can get me a set of intended outputs for these inputs (here's the first one https://github.com/Kong/httpsnippet/blob/7fd160c016d7b00aca923673004dcdcf4b5a0068/test/fixtures/requests/application-form-encoded.json#L1) then I would be happy to do the coding on this.  It's a high desire of mine to have a rust target asap.

**Comment by kevinswiber** [comment]
Which client to use is a significant question.  At this point in time, I believe `reqwest` is the most popular.  https://crates.io/crates/reqwest

**Comment by dimitropoulos** [comment]
Yep, that was the thinking of the insomnia team as well: reqwest.  see https://github.com/Kong/httpsnippet/pull/242

**Comment by SKPG-Tech** [comment]
Any updates on this? I'm really looking forward to Rust snippet generation 😄
